packages/@react-spectrum/cards/test/CardView.test.js

```diff
@@ -0,0 +1,1139 @@
+      expect(within(cards[2]).queryByRole('checkbox')).toBeNull();
+    });
+  });
+
+  describe('loading', function () {
+    it.each`
+      Name                  | layout
+      ${'Grid layout'}      | ${GridLayout}
+      ${'Gallery layout'}   | ${GalleryLayout}
+      ${'Waterfall layout'} | ${WaterfallLayout}
+    `('$Name CardView should render a loading spinner when loading', function ({layout}) {
+      let tree = render(<DynamicCardView layout={layout} items={[]} loadingState="loading" />);
+      act(() => {
+        jest.runAllTimers();
+      });
+
+      let row = tree.getByRole('row');
+      expect(row).toBeTruthy();
+
+      let gridCell = within(row).getByRole('gridcell');
+      expect(gridCell).toBeTruthy();
+
+      let spinner = within(gridCell).getByRole('progressbar');
+      expect(spinner).toHaveAttribute('aria-label', 'Loading…');
+      expect(row.parentNode.style.height).toBe(`${mockHeight}px`);
+
+      tree.rerender(<DynamicCardView layout={layout} />);
+      let grid = tree.getByRole('grid');
+      expect(within(grid).queryByRole('progressbar')).toBeNull();
+      expect(grid).toHaveAttribute('aria-rowcount', defaultItems.length.toString());
+      expect(within(grid).getByText('Title 1')).toBeTruthy();
+    });
+
+    it.each`
+      Name                  | layout
+      ${'Grid layout'}      | ${GridLayout}
+      ${'Gallery layout'}   | ${GalleryLayout}
+      ${'Waterfall layout'} | ${WaterfallLayout}
+    `('$Name CardView should render a loading spinner at the bottom when loading more', function ({layout}) {
+      let tree = render(<DynamicCardView layout={layout} loadingState="loadingMore" />);
+      act(() => {
+        jest.runAllTimers();
+      });
+
+      let grid = tree.getByRole('grid');
+      expect(grid).toHaveAttribute('aria-rowcount', defaultItems.length.toString());
+      expect(within(grid).getByText('Title 1')).toBeTruthy();
+
+      let cards = tree.getAllByRole('gridcell');
+      expect(cards).toBeTruthy();
+      triggerPress(cards[1]);
+
+      act(() => {
+        fireEvent.keyDown(document.activeElement, {key: 'End', code: 35, charCode: 35});
+        fireEvent.keyUp(document.activeElement, {key: 'End', code: 35, charCode: 35});
+        jest.runAllTimers();
+      });
+
+      let spinner = within(grid).getByRole('progressbar');
+      expect(spinner).toHaveAttribute('aria-label', 'Loading more…');
+      expect(getCardStyles(spinner.parentNode).height).toBe('60px');
+    });
+
+    it.each`
+      Name                  | layout
+      ${'Grid layout'}      | ${GridLayout}
+      ${'Gallery layout'}   | ${GalleryLayout}
+    `('$Name CardView should call loadMore when scrolling to the bottom', function ({layout}) {
+      let onLoadMore = jest.fn();
+      let tree = render(<DynamicCardView layout={layout} onLoadMore={onLoadMore} />);
+      expect(tree).toBeTruthy();
+
+       // TODO: loadMore is being called twice on mount??? debug this
```

Comment:
need help?

Reply:
Turns out this is Virtualizer behavior, maybe something to update in the future but test is fixed in follow up PR